File: GO/Cap04/c04ex08.go
// c04ex08.go

package main

import (
  "fmt"
  "strings"
)

func main() {
  var Transp string

  fmt.Println("Escolha o tipo de transporte:")
  fmt.Println()
  fmt.Println("[M] - Motocicleta")
  fmt.Println("[S] - Scooter")
  fmt.Println()

  fmt.Print("==> ")
  fmt.Scan(&Transp)
  fmt.Println()

  Transp = strings.ToUpper(Transp)

  if (Transp == "M" || Transp == "S") {
    fmt.Println("Transporte valido.")
  } else {
    fmt.Println("Transporte invalido.")
  }

  fmt.Println()
  fmt.Print("Tecle <Enter> para encerrar... ")
  fmt.Scanln()
}

File: CPP/Cap04/c04ex08.cpp
// c04ex08.cpp

#include <iostream>
#include <cctype>
using namespace std;

int main(void) {
  char Transp;

  cout << "Escolha o tipo de transporte:\n\n";
  cout << "[M] - Motocicleta\n";
  cout << "[S] - Scooter\n\n";

  cout << "==> ";
  cin >> Transp;
  cin.ignore();

  Transp = toupper(Transp);

  cout << "\n";
  if (Transp == 'M' or Transp == 'S')
    cout << "Transporte valido.\n";
  else
    cout << "Transporte invalido.\n";

  cout << endl;
  cout << "Tecle <Enter> para encerrar... ";
  cin.get();

  return 0;
}
